app/assets/javascripts/discourse/controllers/forgot-password.js.es6

```diff
@@ -52,6 +52,42 @@ export default Ember.Controller.extend(ModalFunctionality, {
       });
 
       return false;
+    },
+
+    emailLogin() {
+      if (this.get('submitDisabled')) return false;
+
+      this.set('disabled', true);
+
+      ajax('/u/email-login', {
+        data: { login: this.get('accountEmailOrUsername').trim() },
```

Comment:
I think we should only accept email here. Allowing the login link to be triggered with just the username may lead to abuse of the feature.